### PR TITLE
Add post-paid billing item support

### PR DESCRIPTION
**Pull request by malteo**

### CODE REVIEW

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java [hunk 1]
```diff
@@ -288,11 +303,29 @@ public void setVmConfiguration( BudgetVMConfigurationDTO vmConfiguration )
...
+    public void setDeferred( Boolean deferred )
+    {
+        this.deferred = deferred;
+    }
+
+    /**
+     * @deprecated by {@link #getDeferred()}
+     */
+    @Deprecated
     public Boolean getDowngrade()
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java [hunk 2]
```diff
@@ -288,11 +303,29 @@ public void setVmConfiguration( BudgetVMConfigurationDTO vmConfiguration )
...
     {
         return downgrade;
     }
 
+    /**
+     * @deprecated by {@link #setDeferred(Boolean)}
+     */
+    @Deprecated
     public void setDowngrade( Boolean downgrade )
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 2 issues
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/enums/OrderType.java [hunk 3]
```diff
@@ -29,5 +29,6 @@
     @ApiModelProperty( "Upgrade disk space, only for managed products" )
     DISK_UPGRADE,
     @ApiModelProperty( "Plan upgrade, only for syndicated products" )
-    SYNDICATED_UPGRADE
+    SYNDICATED_UPGRADE,
+    ESTIMATE
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/enums/OrderType.java]: documenta

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java [hunk 4]
```diff
@@ -38,6 +38,8 @@
     @ApiModelProperty( value = "Billable entries for the estimate", readOnly = true )
     private List<OrderLineDTO> lines = new LinkedList<>();
 
+    private List<OrderLineDTO> postLines;
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java]: ApiModelProperty

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java [hunk 5]
```diff
@@ -38,6 +38,9 @@
     @ApiModelProperty( value = "Billable entries for the estimate", readOnly = true )
     private List<OrderLineDTO> lines = new LinkedList<>();
 
+    @ApiModelProperty( value = "Lines billed at the end of the billing period", readOnly = true )
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/BudgetDTO.java]: MINOR Remove this use of "readOnly"; it is deprecated. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**